[CORE-14465] `rptest`: deflake `CompactionWithRecoveryTest`

WillemKauf opened the pull request:
This could race with a cross-shard (and potentially cross node?) naturally occuring partition movement. An example observed in CI:

```
1. [INFO  - 2025-10-25 05:00:24,458 - partition_movement - cores_converged - lineno:144]: current core placement for topic-monjwggmry/0: [{'node_id': 2, 'core': 2}]
2. [INFO  - 2025-10-25 05:00:25,467 - partition_movement - _dispatch_random_partition_move - lineno:302]: chose 1 replacements for topic-monjwggmry/0: [{'node_id': 2}] -> [{'node_id': 2, 'core': 0}]
3. DEBUG 2025-10-25 05:00:25,541 [shard 0:main] cluster - partition_manager.cc:287 - Log created manage completed, ntp: {kafka/topic-monjwggmry/0}, rev: 45, 17 segments, 2198651 bytes
4. DEBUG 2025-10-25 05:00:27,554 [shard 5:main] cluster - partition_manager.cc:287 - Log created manage completed, ntp: {kafka/topic-monjwggmry/0}, rev: 45, 20 segments, 4767212 bytes
5. [INFO  - 2025-10-25 05:00:29,530 - partition_movement - cores_converged - lineno:144]: current core placement for topic-monjwggmry/0: [{'node_id': 2, 'core': 5}]
6. [INFO  - 2025-10-25 05:00:31,538 - partition_movement - cores_converged - lineno:144]: current core placement for topic-monjwggmry/0: [{'node_id': 2, 'core': 5}]
7. [INFO  - 2025-10-25 05:00:33,546 - partition_movement - cores_converged - lineno:144]: current core placement for topic-monjwggmry/0: [{'node_id': 2, 'core': 5}]
...
```

In step 1, the initial assignment for the partition is observe as core 0. Next, in step 2, core 0 is chosen as the destination for the cross-shard movement for partition 0, and about ~100ms later in step 3, the log is created on shard 0. However, in step 4, the log is moved once again to shard 5, where it is now hosted. In step 5 and beyond, we timeout waiting for the cross-shard transfer to finish with core 0 as the destination, since the log has been moved to shard 5 in the interium and the partition movement request has been stepped upon.

Fixes https://redpandadata.atlassian.net/browse/CORE-14465.

## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

* none

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/75575-->
<!--45ab021c-e99d-4c9c-9aea-e4e5f7f0696f-->
<!--05974815-b35c-4b06-a16f-6718d4102cf8-->
<!--8e6281c5-e7bb-4bab-94f4-1362d3edbc14-->
<!--88de49de-f4ec-43ea-85c2-f85591f15ac4-->
<!--e0b4c85b-8e4b-42b1-b740-e50ae0d96465-->
<!--title-->
## Retry command for [Build#75575](https://buildkite.com/redpanda/redpanda/builds/75575)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/partition_movement_test.py::PartitionMovementTest.test_empty@{"num_to_upgrade":2}
tests/rptest/tests/partition_movement_test.py::PartitionMovementTest.test_empty@{"num_to_upgrade":0}
tests/rptest/transactions/compaction_e2e_test.py::CompactionWithRecoveryTest.test_tx_compaction_with_recovery@{"delete_retention_ms":3}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--3fd997b6-c230-4d8e-b9d1-909d0051806f-->
<!--TEST RESULTS-->
<!--21437532-cc0d-45b1-9490-ff144b41759a-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75583">build#75583</a></summary>


| test_class                      | test_method                                  | test_arguments                                                             | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                             |
|:--------------------------------|:---------------------------------------------|:---------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| SegmentMsTest                   | test_segment_rolling_with_retention_consumer | null                                                                       | integration | https://buildkite.com/redpanda/redpanda/builds/75583#019a5074-ccac-4a65-b002-576acd1c914f | FLAKY         | 19/21    | upstream reliability is '94.7109471094711'. current run reliability is '90.47619047619048'. drift is 4.23476 and the allowed drift is set to 50. The test should PASS   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SegmentMsTest&test_method=test_segment_rolling_with_retention_consumer |
| SIPartitionMovementTest         | test_shadow_indexing                         | {"cloud_storage_type": 1, "num_to_upgrade": 0, "with_cloud_topics": false} | integration | https://buildkite.com/redpanda/redpanda/builds/75583#019a506d-8bb6-4467-a559-b02dbdf16d4f | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SIPartitionMovementTest&test_method=test_shadow_indexing               |
| RedpandaNodeOperationsSmokeTest | test_node_ops_smoke_test                     | {"cloud_storage_type": 1, "mixed_versions": true}                          | integration | https://buildkite.com/redpanda/redpanda/builds/75583#019a5074-ccb3-44a0-8afe-63569d251bae | FLAKY         | 11/21    | upstream reliability is '99.21996879875195'. current run reliability is '52.38095238095239'. drift is 46.83902 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RedpandaNodeOperationsSmokeTest&test_method=test_node_ops_smoke_test   |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/75958">build#75958</a></summary>


| test_class                    | test_method                    | test_arguments                                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                               |
|:------------------------------|:-------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingReplicationTests | test_replication_basic         | {"shuffle_leadership": true, "source_cluster_spec": {"cluster_type": "redpanda"}}                                                                              | integration | https://buildkite.com/redpanda/redpanda/builds/75958#019a6fc1-aff9-45f8-a9b3-8426a86aad67 | FLAKY         | 20/21    | upstream reliability is '91.14331723027375'. current run reliability is '95.23809523809523'. drift is -4.09478 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_basic         |
| ShadowLinkingReplicationTests | test_replication_with_failures | null                                                                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/75958#019a6fc1-aff3-4f43-88ee-81e6e2ba35d1 | FLAKY         | 20/21    | upstream reliability is '96.74952198852772'. current run reliability is '95.23809523809523'. drift is 1.51143 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_with_failures |
| TieredStorageTest             | test_tiered_storage            | {"cloud_storage_type_and_url_style": [2, "virtual_host"], "test_case": {"name": "(TS_Read == True, TS_Timequery == True, SpilloverManifestUploaded == True)"}} | integration | https://buildkite.com/redpanda/redpanda/builds/75958#019a6fba-024a-43f3-8288-fbd15f04ccf1 | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS               | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TieredStorageTest&test_method=test_tiered_storage                        |
</details>
<!--/body-->

WillemKauf committed:
`rptest`: disable autobalancing in `CompactionWithRecoveryTest`

This test utilizes `_do_move_and_verify()` from `PartitionMovementMixin`
directly. Partition autobalancing can cause these primitives to flake.

Disable partition autobalancing to prevent the test from flaking.
